sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -291,6 +292,8 @@ public OfferRequirement getExistingOfferRequirement(PodInstanceRequirement podIn
             taskInfoBuilder.setDiscovery(getDiscoveryInfo(taskSpec.getDiscovery().get(), podInstance.getIndex()));
         }
 
+        taskInfoBuilder.setContainer(Protos.ContainerInfo.newBuilder().setType(Protos.ContainerInfo.Type.MESOS));
```

Comment:
We do already. Adding this proto record only has the effect of telling `navstar` that this is a `MESOS` type container. It's the magic that makes the whole thing work :).